Switch to clang-format-10 to support more recent platforms

tautschnig opened the pull request:
clang-format-7 (or older) might not even be available anymore on some
systems, and will certainly not be the default. clang-format-10 has
improved support for initializer lists, resulting in different
formatting of these:
```
foo bar{value_1,
        value_2};
```
now is formatted as
```
foo bar{
  value_1,
  value_2};
```
which is more consistent with how the same would have been formatted
had parentheses been used.

clang-format-10 is even available on Ubuntu 18.04, which is the most
dated system our CI builds are still running on.

Fixes: #6457

<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.

If your PR fixes a bug, include the regression test(s) in the same commit as the bug fix. Else, keep commits small and orthogonal, possibly placing tests in commits of their own.
--->

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [x] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

tautschnig committed:
Switch to clang-format-10 to support more recent platforms

clang-format-7 (or older) might not even be available anymore on some
systems, and will certainly not be the default. clang-format-10 has
improved support for initializer lists, resulting in different
formatting of these:
```
foo bar{value_1,
        value_2};
```
now is formatted as
```
foo bar{
  value_1,
  value_2};
```
which is more consistent with how the same would have been formatted
had parentheses been used.

clang-format-10 is even available on Ubuntu 18.04, which is the most
dated system our CI builds are still running on.

Fixes: #6457

NlightNFotis reviewed (approved):
Seeing this made me so excited (I got bit by this just yesterday) that I was looking for approval GIFs but Github's GIF integration doesn't look that great.

In any case, thank you for handling that. It is so frustrating to have clang-format do one thing locally and another remotely.

thomasspriggs reviewed (approved):
Seems worth noting that merging this PR is likely to cause mild consternation for anyone who has an open PR or work in progress branch which needs reformatting. It should be straight forward to resolve for these individuals by updating their local clang-format versions and running `git-clang-format` over the branch. The reformatting can be squashed into individual commits in a branch using `git filter-branch --tree-filter 'git-clang-format develop' -- develop..HEAD`.

Apart from the above note, I am happy to see this version updated.

tautschnig committed:
Use github.com Linux kernel mirror in integration test

CI jobs running on github are less likely to temporarily fail when using
mirrors from github.com. The (transient?) failure seen most recently
was:
```
git clone -b v5.10 --depth=1 https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux-stable.git linux_5_10
Cloning into 'linux_5_10'...
remote: aborting due to possible repository corruption on the remote side.
fatal: early EOF
fatal: fetch-pack: invalid index-pack output
```
This commit seeks to address this kind of problem.
